src/Bicep.Core/Features/IFeatureProviderManager.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+
+namespace Bicep.Core.Features;
+
+public interface IFeatureProviderManager
+{
+    IFeatureProvider GetFeatureProvider(Uri templateUri);
+
+    static IFeatureProviderManager ForFeatureProvider(IFeatureProvider featureProvider) 
+        => new ConstantFeatureProviderManager(featureProvider);
+
+    private class ConstantFeatureProviderManager : IFeatureProviderManager
+    {
+        private readonly IFeatureProvider features;
+
+        internal ConstantFeatureProviderManager(IFeatureProvider features)
+        {
+            this.features = features;
+        }
+
+        public IFeatureProvider GetFeatureProvider(Uri templateUri) => features;
```

Comment:
I couldn't find any concrete implementations where the `templateUri` parameter is actually used. How does this manager class differentiate between configuration in different locations?

Reply:
My intention was to address #7955 in a follow-up PR, though I can roll that into this one (since the switch to using IFeatureManagerProviders is necessary for pointless without that change).

Reply:
On second thought, the IFeatureProvider -> IFeatureProviderManager changes don't make sense w/o the #7955 work, which I had left out of this PR for brevity's sake. I added that in as a single commit so that it can be pulled out if you think that makes the PR too big.

Reply:
On further reflection, the work for #7955 adds too much extra functionality to this PR and makes it difficult to review. I'm going to split everything related to `IFeatureProvider`s into a separate PR so that this one just addresses the config loading bug.

Reply:
Thanks - sorry for the randomization. This PR is definitely already pretty large, so great thought to move to a separate PR.